dev/sg/sg_update.go

```diff
@@ -3,18 +3,26 @@ package main
 import (
 	"context"
 	"flag"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
 	"os/exec"
+	"runtime"
+	"strings"
 
 	"github.com/peterbourgon/ff/v3/ffcli"
 
 	"github.com/sourcegraph/sourcegraph/dev/sg/internal/run"
 	"github.com/sourcegraph/sourcegraph/dev/sg/internal/stdout"
+	"github.com/sourcegraph/sourcegraph/lib/errors"
 	"github.com/sourcegraph/sourcegraph/lib/output"
 )
 
 var (
-	updateFlags   = flag.NewFlagSet("sg update", flag.ExitOnError)
-	updateToLocal = updateFlags.Bool("local", false, "Update to local copy of 'dev/sg'")
+	updateFlags    = flag.NewFlagSet("sg update", flag.ExitOnError)
+	updateToLocal  = updateFlags.Bool("local", false, "Update to local copy of 'dev/sg'")
+	updateDownload = updateFlags.Bool("download", false, "Download instead of building")
```

Comment:
Nitpick: I think `use-prebuilt` would be easier to understand. But that truly is a nitpick, feel free to ignore.

But I'd extend the description a bit:

```suggestion
	updateDownload = updateFlags.Bool("download", false, "Download a prebuilt binary of 'sg' instead of compiling it locally")
```